### DISCUSSION TITLE
[2023-04-03] Library and Engine Team Update

### DISCUSSION THREAD

**Discussion by jdunkerley** [enso-org/enso#6186] [Weekly Demo Day]
- James: Adjusted column resizing within the table to be data content driven. #6157 
- James: Adjusted table header function to not adjust the header's case. #6160
- James: Altered `Process.run` to return standard-out, standard-error, and exit code. #6160 
- James: No warning attached if joining data and just adding the prefix. #6160 
- James: Design discussion for Map literals. #6167
- James: Sorting out type definitions and reviewing public/private functions.
- Radek: Completed adding the initial Value_Type implementation to tables. #6073 
- Radek: Added proper datetime and time support to Postgres. #6115 
- Radek: Added year, month and day to Column type. #6115 
- Radek: Started on `Table.union` support within a database, created a prototype.
- Greg: Completed updating `Text.replace` and `Text.replace_all`. #5959 
- Greg: Updated `Text.split` to new API. #6116 
- Greg: Adding `Text.tokenize` for regular expression-based splitting. #6150 
- Greg: Adjust `Text.split` to allow a collection of delimiters. #6156 
- Jaroslav: Fix for parser NPE on empty assignment. #6107 
- Jaroslav: Always print stacktrace to provide some guidance of what is going on. #6128 
- Jaroslav: Restoring disabled parser tests. #6143
- Jaroslav: Delay creation `enso_project`. #6149 
- Jaroslav: Lazy atoms PoC. #6151 
- Jaroslav: `Text.to_display_text` is a shortened version of self. #6174 
- Jaroslav: Twice as fast `Any.==`. #6184
- Hubert: Regenerate the Index when building standard libraries. #6000 
- Hubert: More elaborate error messages when text edit fails. #6108
- Hubert: Missing imports sometimes result in very confusing error traces. #6123 
- Hubert: Working on the Runtime.Context and methods to handle it. #6171 
- Pavel M: Fixing the issue where `Boolean.False != False`. #6090 
- Pavel M: Continued work on sort handling Incomparable types by default. #5998 
- Dmitry: Reviewed that editing a node no longer produces masses of data flow errors. #5072 
- Dmitry: When renaming the project clean old modules instead of updating. #6148 
- Dmitry: Batching insert suggestions into the database.
- Pawel G: Working to replace node text label with widget hierarchy. #5921

**Comment by 4e6** [comment]
Batch insert suggestions improved suggestions loading by ~3 times (1582 ms before vs 458 ms after on screenshots below)

![2023-04-03-173423_2086x324_scrot](https://user-images.githubusercontent.com/357683/229553898-2b6375c0-883d-4a67-97a6-d61fd99f86ed.png)
![2023-04-03-173514_2083x321_scrot](https://user-images.githubusercontent.com/357683/229553905-cd5a0dc4-46e7-4ce7-beff-7908133094c8.png)
